logger.ts

```diff
@@ -144,15 +142,47 @@ class Logger {
     fileStream: fs.WriteStream
   ) {
     for await (const chunk of message) {
-      if (!fileStream.write(chunk)) {
-        await once(fileStream, "drain");
-      }
+      const pipeline = promisify(stream.pipeline);
+      pipeline(chunk, fileStream).catch((err) => {
+        console.error("Pipeline failed.", err);
+        fileStream.end();
+      });
     }
   }
 
-  private _formatMessageForFile(message: string, prefix: string): string {
+  public formatMessageForFile(message: string, prefix: string): string {
     return prefix + message + "\n";
   }
+
+  private async _logToDestination(
+    message: string,
+    prefix: string,
+    logDestination?: LogDestination,
+    logFileName?: string
+  ) {
+    if (this._isLogToConsole(logDestination)) {
+      this._logToConsoleWrapper(message, prefix);
+    } else {
+      await this._logToFile(message, prefix, logFileName);
+    }
+  }
+  // This is done for testing purposes
+  private _logToConsoleWrapper(message: string, prefix: string) {
+    switch (prefix) {
+      case this.logPrefix.debug:
+        console.debug(message);
```

Comment:
here to use:
```suggestion
        console[level](message);
```
and delete all the others